public/remote/js/controllers/mainController.js

```diff
@@ -160,6 +125,44 @@ app.controller("mainController", function($scope, $rootScope, config, $routePara
     $scope.setCurrentVideo = function(video){
         $scope.current.video = video;
 
+
+        // Load all related overlays
+        $overlayService.list_by_video($scope.current.video.video_id)
+        .then(function onSuccess(response) {
+            // Make sure the overlays are in the scenario
+            $scope.filter = {};
+            $scope.filter.relationship_type = "overlay";
+            $relationshipService.list_by_label("belongs_to", $scope.pagination, $scope.filter)
+                .then(function(responseBelongsTo){
+                    $scope.filter = undefined;
+                    $scope.overlays = [];
+                    for(let i = 0; i < response.data.length; i++){
+                        for(let k = 0; k < responseBelongsTo.data.length; k++){
+                            if(response.data[i].overlay_id === responseBelongsTo.data[k].overlay_id && responseBelongsTo.data[k].scenario_id === $scope.current.scenario.scenario_id){
+                                let exists = false;
+                                if($scope.overlays.length > 0){
+                                    for(let j = 0; j < $scope.overlays.length; j++){
+                                        if($scope.overlays[i] === response.data[i]){
+                                            exists = true;
+                                        }
+                                        if(j === $scope.overlays.length - 1 && exists === false){
+                                            $scope.overlays.push(response.data[i]);
+                                        }
+                                    }
+                                }
+                                else{
+                                    $scope.overlays.push(response.data[i]);
+                                }
+                            }
+                        }
+                    }
+                }).catch(function onError(responseBelongsTo) {
+                    $scope.err = responseBelongsTo.data;
+                });
```

Comment:
Nesting multiple promise chains and having multiple `.catch()` handlers usually is not necessary and bad style.
Untangling this nested promise, so as a first step I'd just get rid of this catch, and let it be handled by the outer/lower catch handler. This is achieved by creating a proper Promise chain: Just return the inner promise in the outer promise handler (see suggestion for line 135
```suggestion
                });
```